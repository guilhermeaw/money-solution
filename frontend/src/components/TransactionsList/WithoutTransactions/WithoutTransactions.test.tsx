import { render, screen } from '@testing-library/react';

import { WithoutTransactions } from '.';

const createComponent = () => {
  return render(<WithoutTransactions />);
};

describe('WithoutTransactions component', () => {
  it('should render correctly', () => {
    const { container } = createComponent();

    expect(
      screen.getByAltText(/Sem transações cadastradas/i),
    ).toBeInTheDocument();

    expect(
      screen.getByRole('heading', { name: /Que pena.../i }),
    ).toBeInTheDocument();

    expect(
      screen.getByText(
        /Parece que você ainda não possui transações cadastradas/i,
      ),
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
