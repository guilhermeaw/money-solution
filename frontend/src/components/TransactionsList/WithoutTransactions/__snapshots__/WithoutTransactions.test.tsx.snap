// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithoutTransactions component should render correctly 1`] = `
<div
  class="MuiStack-root-fLKvZh iZsUYB"
>
  <img
    alt="Sem transações cadastradas"
    src="/assets/images/no-transactions.svg"
    width="400px"
  />
  <h2
    class="MuiTypography-root-bXavBU kJcdFH MuiTypography-root MuiTypography-h2"
  >
    Que pena...
  </h2>
  <h6
    class="MuiTypography-root-bXavBU dWiFMB MuiTypography-root MuiTypography-subtitle1"
  >
    Parece que você ainda não possui transações cadastradas
  </h6>
</div>
`;
